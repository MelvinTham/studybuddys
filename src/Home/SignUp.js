import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import TextField from '@material-ui/core/TextField';
import Button from '@material-ui/core/Button';
import Typography from '@material-ui/core/Typography';
import Card from '@material-ui/core/Card';
import './Home.css';
import * as firebase from 'firebase';

class SignUp extends Component {
    constructor(props) {
        super(props);
        this.state = {
            name:'',
            email:'',
            password:'',
            checkPassword:''

        }
        this.handleChange = this.handleChange.bind(this);
        this.createUser = this.createUser.bind(this);
        this.checkLoggedIn = this.checkLoggedIn.bind(this);
    }

    handleChange = name => event => {
        this.setState({
            [name]: event.target.value
        });
    }

    createUser(e){
      e.preventDefault();
      var userName = this.state.name;
      var userEmail = this.state.email;
      var prop = this.props;
      if(this.state.checkPassword === this.state.password){
         firebase.auth().createUserWithEmailAndPassword(userEmail, this.state.password)
         .then(function(){
            var user = firebase.auth().currentUser;
            firebase.database().ref('users/' + user.uid).set({
               email: userEmail,
               name : userName,
               courseList: ""
            })
            .then(function(){
               prop.history.push('/dashboard');
            })
            .catch(function (error) {
               alert(error.message);
            });
         })
         .catch(function (error) {
            alert(error.message);
         });
      }
      else{
         alert("Passwords do not match");
      }
    }

    render() {
        return (
            <div data-aos = "" className = "Home">
                <Card
                    raised = {true}
                    className = "signUp"
                    data-aos="fade-down"
                    data-aos-easing="linear"
                    data-aos-duration="500">
                <Typography
                    component="h2"
                    variant="display2"
                    gutterBottom = {true}
                    style={{color:'black'}}
                >
                    Sign Up
                </Typography>
                    <form>
                        < TextField className = ""
                            placeholder = "Full Name"
                            onChange={this.handleChange('name')}
                            />

                        <br/>

                        < TextField
                            className = ""
                            type = 'email'
                            placeholder = "Email Address"
                            style = {{marginTop:'15px'}}
                            onChange={this.handleChange('email')}
                        />

                        <br/>

                        < TextField
                            type = "password"
                            className = ""
                            placeholder = "Password"
                            style = {{marginTop:'15px'}}
                            onChange={this.handleChange('password')}
                        />

                        <br/>

                        < TextField
                            type = "password"
                            className = ""
                            placeholder = "Confirm Password"
                            style = {{marginTop:'15px'}}
                            onChange={this.handleChange('checkPassword')}
                        />

                        <br/>

                        <Button
                            onClick={this.createUser}
                            type = 'submit'
                            variant = "outlined"
                            style = {{marginTop:'15px'}}
                        >
                            Sign Me Up
                        </Button>

                    </form>

                    <Link
                        to = '/'
                        variant = "outlined"
                        style = {{marginTop:'15px'}}
                    >
                        Have an account? Log In Here
                    </Link>
                </Card>
            </div>
        );
    }
}

export default SignUp;
