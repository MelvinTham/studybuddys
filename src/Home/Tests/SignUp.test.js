import Enzyme, { render, shallow } from 'enzyme';
import enzymeAdapterReact16 from 'enzyme-adapter-react-16';
import React from 'react';
import SignUp from './../SignUp';
import { BrowserRouter as Router } from 'react-router-dom';

Enzyme.configure({ adapter: new enzymeAdapterReact16() });
const getDefaultProps = () => ({});
describe('SignUp component', () => {
  it('shallow renders without crashing', () => {
    const {} = getDefaultProps();
    shallow(
        <Router>
            <SignUp />
        </Router>);
  });
  it('render snapshot', () => {
    const {} = getDefaultProps();
    const wrapper = render(
        <Router>
            <SignUp />
        </Router>);
    expect(wrapper).toMatchSnapshot();
  });
});
