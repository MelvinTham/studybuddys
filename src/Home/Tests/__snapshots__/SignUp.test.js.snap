// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp component render snapshot 1`] = `
<div
  class="Home"
  data-aos=""
>
  <div
    class="MuiPaper-root-2 MuiPaper-elevation8-12 MuiPaper-rounded-3 MuiCard-root-1 signUp"
    data-aos="fade-down"
    data-aos-duration="500"
    data-aos-easing="linear"
  >
    <h2
      class="MuiTypography-root-29 MuiTypography-display2-32 MuiTypography-gutterBottom-56"
      style="color:black"
    >
      Sign Up
    </h2>
    <form>
      <div
        class="MuiFormControl-root-64"
      >
        <div
          class="MuiInputBase-root-81 MuiInput-root-68 MuiInput-underline-72 MuiInputBase-formControl-82 MuiInput-formControl-69"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input-91 MuiInput-input-76"
            placeholder="Full Name"
            type="text"
          />
        </div>
      </div>
      <br />
      <div
        class="MuiFormControl-root-64"
        style="margin-top:15px"
      >
        <div
          class="MuiInputBase-root-81 MuiInput-root-68 MuiInput-underline-72 MuiInputBase-formControl-82 MuiInput-formControl-69"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input-91 MuiInput-input-76 MuiInputBase-inputType-94 MuiInput-inputType-79"
            placeholder="Email Address"
            type="email"
          />
        </div>
      </div>
      <br />
      <div
        class="MuiFormControl-root-64"
        style="margin-top:15px"
      >
        <div
          class="MuiInputBase-root-81 MuiInput-root-68 MuiInput-underline-72 MuiInputBase-formControl-82 MuiInput-formControl-69"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input-91 MuiInput-input-76 MuiInputBase-inputType-94 MuiInput-inputType-79"
            placeholder="Password"
            type="password"
          />
        </div>
      </div>
      <br />
      <div
        class="MuiFormControl-root-64"
        style="margin-top:15px"
      >
        <div
          class="MuiInputBase-root-81 MuiInput-root-68 MuiInput-underline-72 MuiInputBase-formControl-82 MuiInput-formControl-69"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input-91 MuiInput-input-76 MuiInputBase-inputType-94 MuiInput-inputType-79"
            placeholder="Confirm Password"
            type="password"
          />
        </div>
      </div>
      <br />
      <button
        class="MuiButtonBase-root-124 MuiButton-root-98 MuiButton-outlined-106"
        style="margin-top:15px"
        tabindex="0"
        type="submit"
      >
        <span
          class="MuiButton-label-99"
        >
          Sign Me Up
        </span>
      </button>
    </form>
    <a
      href="/"
      style="margin-top:15px"
      variant="outlined"
    >
      Have an account? Log In Here
    </a>
  </div>
</div>
`;
